flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/PartitionInfo.java

```diff
@@ -18,48 +18,64 @@
 
 package org.apache.flink.runtime.executiongraph;
 
-import org.apache.flink.runtime.deployment.InputChannelDeploymentDescriptor;
+import org.apache.flink.runtime.clusterframework.types.ResourceID;
+import org.apache.flink.runtime.deployment.TaskDeploymentDescriptorFactory;
 import org.apache.flink.runtime.jobgraph.IntermediateDataSetID;
-import org.apache.flink.util.Preconditions;
+import org.apache.flink.runtime.shuffle.ShuffleDeploymentDescriptor;
 
+import javax.annotation.Nonnull;
 import java.io.Serializable;
 
 /**
  * Contains information where to find a partition. The partition is defined by the
- * {@link IntermediateDataSetID} and the partition location is specified by
- * {@link InputChannelDeploymentDescriptor}.
+ * {@link IntermediateDataSetID} and the partition is specified by
+ * {@link org.apache.flink.runtime.shuffle.ShuffleDeploymentDescriptor}.
  */
 public class PartitionInfo implements Serializable {
 
 	private static final long serialVersionUID = 1724490660830968430L;
 
+	@Nonnull
 	private final IntermediateDataSetID intermediateDataSetID;
```

Comment:
I'm wondering whether the `ShuffleDeploymentDescriptor` should not be able to identify itself (namely which partition it describes) returning the `IntermediateResultPartitionID` and `IntermediateDataSetID`.

Reply:
At the moment, `IntermediateDataSetID` is kept once in `InputGateDeploymentDescriptor` for all channels/partitions as before. I suggest we keep it this way in this PR. Later we can consider one step further refactoring where `IntermediateDataSetID` is part of the full partition id.